envoy/tests/legacy/test_envoy.py

```diff
@@ -133,15 +133,17 @@ def test_unknown(fixture_path, mock_http_response, dd_run_check, check):
 
 @pytest.mark.unit
 @pytest.mark.parametrize(
-    'test_case, extra_config, expected_http_kwargs',
+    'extra_config, expected_http_kwargs',
     [
-        ("new auth config", {'username': 'new_foo', 'password': 'new_bar'}, {'auth': ('new_foo', 'new_bar')}),
-        ("legacy ssl config True", {'verify_ssl': True}, {'verify': True}),
-        ("legacy ssl config False", {'verify_ssl': False}, {'verify': False}),
-        ("legacy ssl config unset", {}, {'verify': True}),
+        pytest.param(
```

Comment:
For educational purposes: what's the purpose of this change?

Reply:
Here we were adding one parameter to the test that is not used and that is there only to "name" the test itself. On master, you have something like that when running the tests:

```
tests/legacy/test_envoy.py::test_config[new auth config-extra_config0-expected_http_kwargs0] PASSED [ 20%]
tests/legacy/test_envoy.py::test_config[legacy ssl config True-extra_config1-expected_http_kwargs1] PASSED [ 21%]
tests/legacy/test_envoy.py::test_config[legacy ssl config False-extra_config2-expected_http_kwargs2] PASSED [ 23%]
tests/legacy/test_envoy.py::test_config[legacy ssl config unset-extra_config3-expected_http_kwargs3] PASSED [ 24%]
```

Using pytest.param allow you to specify the `id` of the test without having an extra parameter to the test itself. You endup having something like this when you are running the tests: 

```
tests/legacy/test_envoy.py::test_config[new auth config] PASSED          [ 20%]
tests/legacy/test_envoy.py::test_config[legacy ssl config True] PASSED   [ 21%]
tests/legacy/test_envoy.py::test_config[legacy ssl config False] PASSED  [ 23%]
tests/legacy/test_envoy.py::test_config[legacy ssl config unset] PASSED  [ 24%]
```

Which only shows the id of the test, and avoid having an unused `test_case` parameter